### DISCUSSION TITLE
docs/config/replace

### DISCUSSION THREAD

**Discussion by giscus** [STranslate/STranslate#202] [Announcements]
# docs/config/replace

WPF 开发的一款即开即用、即用即走的翻译(OCR)工具

https://stranslate.zggsong.com/docs/config/replace

**Comment by qq50608** [comment]
工程师您好，使用中遇到了两个问题，需请教下。1：替换翻译时，如何取消最前面出现的“翻”字或自定义？2：免费翻译源内的Deepl，经常显示too many 什么，就是请求太多的意思，除了使用自己的KEY，还有什么解决方式呢？出现的太频繁了。

**Reply by ZGGSONG** [reply]
1. 不支持自定义，而且底层实现在不同编辑器里表现不同，避免出现问题，不会开放自定义
2. 公共服务访问人数太多，建议自建服务，如果是想用deepl，直接使用stranslate(本地)服务即可

**Reply by qq50608** [reply]
收到，感谢。那么通过热键进行替换翻译的时候，是否可以取消“翻”字的提示呢？或者改成个“翻_”，或者“翻 译（中间空格）”，仅仅是个建议，感觉这样的话，后期用户可以自己批量进行删除，如果只是“翻"，会有很多冲突。               顺祝商祺。

**Reply by ZGGSONG** [reply]
你好，我有个疑问，我不太明白，这个"翻"引发了哪些问题，这个可以详细描述一下，如果有无法避免的问题，我这边可以处理一下。

**Reply by Yiyang-Sun** [reply]
我同样对此有些困惑: 为什么需要在翻译后的文本前加入一个"翻"字? 这样做的目的是什么?不是应该只呈现翻译后的文本吗?

**Reply by ZGGSONG** [reply]
触发替换翻译后，首先获取原始文本，随后填充翻译中...表明状态，随后拿到翻译结果后删除该文本后模拟键盘输入。如果有删的不全的情况，可以详细说明一下，在什么系统、软件、软件版本下出现了该情况

**Reply by Yiyang-Sun** [reply]
系统版本: Win11 Pro, 24H2, OS build: 26100.2152
Quicker版本: 1.43.33
PowerPoint版本: Office 365订阅, version 2402
存在删不全的情况, 会剩余一个"翻"字.

**Reply by ZGGSONG** [reply]
行吧 各种软件各种适配问题  我设计个新方案替代一下吧

**Reply by ZGGSONG** [reply]
![动画](https://github.com/user-attachments/assets/534e4b9e-3b4b-4ef8-9320-1442d3c60d43)

**Reply by Yiyang-Sun** [reply]
请问这个新的方案已经更新到1.2.6了吗?

**Reply by ZGGSONG** [reply]
下个版本发布

**Comment by huouer** [comment]
什么是“替换翻译”？

**Reply by ZGGSONG** [reply]
选中文本翻译成目标语言后替换掉原有文本

**Comment by Yiyang-Sun** [comment]
请问, 替换翻译如果选择OpenAI, 是否会调用"翻译服务"中的配置的OpenAI prompt?
多谢!

**Reply by ZGGSONG** [reply]
会的
